### PR TITLE
add script to build Ubuntu cross compilation toolchain

### PR DESCRIPTION
**Pull request by weissi**
To cross compile a binary, you will need a cross compilation toolchain which consists of the toolchain itself and an SDK. For example compiling an iPhone binary from your Mac will use `/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.3.sdk` or similar. If you want to compile a Swift binary for say Ubuntu Xenial on your Mac, you'll need the following pieces of software

 - a compiler being able to cross compile ✅ (`swiftc` and `clang` are able to do that)
 - the basic Swift libraries (like `libswiftCore`) for Ubuntu ✅ (that's part of the Swift distribution for Ubuntu)
 - an SDK for Ubuntu Linux ✅ (that can be assembled from the `.deb` packages available from Ubuntu)
 - an ELF linker that runs on macOS ✅ (the [gold linker](https://sourceware.org/binutils/) from the GNU binutils can be compiled on macOS)

In other words, all the pieces of software are available for free online, they just need to be assembled in the correct way. This script is an example on how you can do that for Ubuntu Xenial.

Example use:

    Utilities/build_ubuntu_cross_compilation_toolchain /tmp/ ~/Downloads/swift-3.1-RELEASE-osx.pkg ~/Downloads/swift-3.1-RELEASE-ubuntu16.04.tar.gz

This should (after quite some time) result in a file called `/tmp/cross-toolchain/ubuntu-xenial-destination.json` which you can hand to SwiftPM using the `--destination` flag introduced in #1098 . An example for that is

    cd my-spm-project
    swift build --destination /tmp/cross-toolchain/ubuntu-xenial-destination.json

which should result in a binary for Linux 😃

    $ file .build/debug/MySPMTestProgram
    .build/debug/SwiftHttpServerTest: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, not stripped

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
I think we should add the usage instructions somewhere, maybe at top of the file or as a separate markdown file.

**Comment by weissi** [comment]
added that to the `usage` output of the script itself.

File: Utilities/build_ubuntu_cross_compilation_toolchain [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        echo "$1"
+    else
+        (
+        cd "$(dirname "$1")"
+        echo "$(pwd)/$(basename "$1")"
+        )
+    fi
+}
+
+function fix_glibc_modulemap() {
```
Comment (ddunbar) [inline Utilities/build_ubuntu_cross_compilation_toolchain]: Is this something we expect to always have to do, or should there be a bug tacking eliminating the need for it? (i.e., is it something we fix in Swift)

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

File: Utilities/build_ubuntu_cross_compilation_toolchain (hunk 1 above)
Comment (weissi) [inline Utilities/build_ubuntu_cross_compilation_toolchain]: @ddunbar that's a very good question. I think that's a bug in Swift.

**Pull request merged**